tests/pact/hash.repl

```diff
@@ -25,27 +31,39 @@
 ; pre fork module hashing
 (insert-hash "a" (hash my-mod))
 (insert-hash "b" (hash my-mod))
+(insert-hash "c" (hash [my-mod, {'a:my-mod}, (create-user-guard (f my-mod))]))
```

Comment:
this covers all cases recursively for `my-mod`.

`PactValues` are:
- lists
- objects
- guards (can contain them recursively too)
- literals
- modrefs

Literals cannot contain any pactvalues, so it's just the other 3 cases we must check.